cli/tools/fmt.rs

```diff
@@ -521,6 +535,138 @@ pub fn format_html(
   })
 }
 
+/// A function for formatting embedded code blocks in JavaScript and TypeScript.
+fn create_external_formatter_for_typescript(
+  unstable_options: &UnstableFmtOptions,
+) -> impl Fn(
+  MediaType,
+  String,
+  &dprint_plugin_typescript::configuration::Configuration,
+) -> Option<String> {
+  let unstable_sql = unstable_options.sql;
+  move |media_type, text, config| {
+    match media_type {
+      MediaType::Css => format_embedded_css(&text, config),
+      MediaType::Html => format_embedded_html(&text, config),
+      MediaType::Sql => {
+        if unstable_sql {
+          format_embedded_sql(&text, config)
+        } else {
+          None
+        }
+      }
+      // dprint-plugin-typescript only detects css, html, and sql for now
+      _ => unreachable!(),
+    }
+  }
+}
+
+/// Formats embedded CSS code blocks in JavaScript and TypeScript.
+///
+/// This function supports properties only CSS expressions, like:
+/// ```css
+/// margin: 10px;
+/// padding: 10px;
+/// ```
+///
+/// To support this scenario, this function first wraps the text with `a { ... }`,
+/// and then strips it off after formatting with malva.
+fn format_embedded_css(
+  text: &str,
+  config: &dprint_plugin_typescript::configuration::Configuration,
+) -> Option<String> {
+  use malva::config;
+  let options = config::FormatOptions {
+    layout: config::LayoutOptions {
+      indent_width: config.indent_width as usize,
+      use_tabs: config.use_tabs,
+      ..Default::default()
+    },
+    ..Default::default()
+  };
+  // Wraps the text in a css block of `a { ... }`
+  // to make it valid css (scss)
+  let Ok(text) = malva::format_text(
+    &format!("a{{\n{}\n}}", text),
+    malva::Syntax::Scss,
+    &options,
+  ) else {
+    return None;
+  };
+  let mut buf = vec![];
+  for (i, l) in text.lines().enumerate() {
+    // skip the first line (a {)
+    if i == 0 {
+      continue;
+    }
+    // skip the last line (})
+    if l.starts_with("}") {
+      continue;
+    }
+    let mut chars = l.chars();
+    // drop the indentation
+    for _ in 0..config.indent_width {
+      chars.next();
+    }
+    buf.push(chars.as_str());
+  }
+  Some(buf.join("\n").to_string())
+}
+
+/// Formats the embedded HTML code blocks in JavaScript and TypeScript.
+fn format_embedded_html(
+  text: &str,
+  config: &dprint_plugin_typescript::configuration::Configuration,
+) -> Option<String> {
+  use markup_fmt::config;
+  let options = config::FormatOptions {
+    layout: config::LayoutOptions {
+      indent_width: config.indent_width as usize,
+      use_tabs: config.use_tabs,
+      ..Default::default()
+    },
+    ..Default::default()
```

Comment:
I think we should avoid using `...Default::default()` here and instead set the defaults explicitly. Otherwise the default could change in the plugins and that could break us (same above).